### DISCUSSION TITLE
Cannot get the synchronization work

### DISCUSSION THREAD

**Discussion by veritatis** [TriliumNext/Trilium#1372] [Q&A]
Trilium is exactly the note app I dreamed of. It satisfied almost all my needs. 

However, I just cannot not get the synchronization work. I have a VPS running CentOS and I installed the docker version of trilium server on it. (I didn't change the config file and I use HTTP as default). Then I add the server info in options of a Windows version trilium. Sync test seems to be good. 
![test sync1](https://user-images.githubusercontent.com/33884471/97834944-fbbd3f80-1d13-11eb-931b-108e6af2724d.png)

However, in a different windows pc, I downloaded trillium and chose sync from server, and it failed with msg "Sync setup failed: Request to get http://my VPS IP:8080/api/setup/sync-seed failed, error: 500 Internal Privoxy Error"
![sync from server failed](https://user-images.githubusercontent.com/33884471/97885318-28984380-1d62-11eb-81c3-44793779dbf9.png)

What's wrong with my sync setup?  Thanks a lot!

**Comment by good-praxis** [comment]
Let's debug this, make sure you have access to the server in question in your browser on the problem computer.
The screenshot shows `ERR_CONNECTION_TIMED_OUT` instead of the `500 Internal Privoxy Error` you described, so that leads me to believe it's just not accessible from the other computer right now

Also that second post still contains an IP, you were pretty strict on censoring it in other places so I assume that wasn't intentional, just FYI

I personally had a more straight-forward experience of setting up the server via the [packaged server method](https://github.com/zadam/trilium/wiki/Packaged-server-installation).

**Reply by veritatis** [reply]
Thank you for the information and reminder! I've updated the screenshot. I tried several times and in most of them it's the 500 Internal Privoxy Error. 

I just open the log file and get this
`03:18:11.183 CPU model: Intel Core Processor (Haswell, no TSX, IBRS), logical cores: 1 freq: 2399 Mhz
03:18:11.186 Listening on port 8080
04:58:42.093 DB size: 4 KB
04:58:43.697 DB not initialized, please visit setup page - http://[your-server-host]:8080 to see instructions on how to initialize Trilium.
04:58:43.698 App HTTP server starting up at port 8080
04:58:43.698 {
  "appVersion": "0.45.2",
  "dbVersion": 170,
  "syncVersion": 16,
  "buildDate": "2020-10-29T22:57:25+01:00",
  "buildRevision": "4f75b6aaafef8144080fd17f403a605f61f5590d",
  "dataDirectory": "/root/trilium-data",
  "clipperProtocolVersion": "1.0"
}` 

and I went to the setup page to find the following

`500 Internal Privoxy Error
Privoxy encountered an error while processing your request:
Could not load template file no-server-data or one of its included components.
Please contact your proxy administrator.
If you are the proxy administrator, please put the required file(s)in the (confdir)/templates directory. The location of the (confdir) directory is specified in the main Privoxy config file. (It's typically the Privoxy install directory).` 

 
I'm a Linux noob and all I can do is to copy/paste commands that I find online into my SSH client without truly understanding them. I tried the packaged server method earlier but also failed. I'll try it again.

**Reply by good-praxis** [reply]
On the computer on which sync works you should be able to check out the server logs 
![image](https://user-images.githubusercontent.com/22496800/97887207-e05d2f00-1d29-11eb-813b-db79a113c7d9.png)
They might hold a clue as to what is happening

**Reply by good-praxis** [reply]
Be sure to do the above directly on the web, not in the application client

**Reply by good-praxis** [reply]
As for the packaged server, the docs a bit outdated, you'll have to extract the current packaged folder via `tar -xf [file]` instead of `p7zip`

**Reply by veritatis** [reply]
> As for the packaged server, the docs a bit outdated, you'll have to extract the current packaged folder via `tar -xf [file]` instead of `p7zip`

Thank you so much! I removed the docker installation and used the `tar -xf` command and it just worked. Previously I used p7zip and was told that the extraction was succeed, but then the installation failed.
